src/physlr-molecules.cc

```diff
@@ -0,0 +1,412 @@
+	std::cerr << "in sec: " << omp_get_wtime() - sTime << std::endl;
+	sTime = omp_get_wtime();
+#endif
+
+	std::cerr << "Memory usage: " << double(memory_usage()) / double(1048576) << "GB" << std::endl;
+}
+
+void
+biconnectedComponents(graph_t& subgraph, vertexToComponent_t& vertexToComponent)
+{
+	// Find biconnected components
+	boost::property_map<graph_t, edgeComponent_t>::type component =
+	    boost::get(edgeComponent, subgraph);
+	boost::biconnected_components(subgraph, component);
+
+	std::vector<vertex_t> artPointsVec;
+	articulation_points(subgraph, std::back_inserter(artPointsVec));
+	std::unordered_set<vertex_t> art_points(artPointsVec.begin(), artPointsVec.end());
+
+	// Remove articulation points from biconnected components
+	boost::graph_traits<graph_t>::edge_iterator ei, ei_end;
+	componentToVertexSet_t componentToVertexSet;
+
+	for (boost::tie(ei, ei_end) = boost::edges(subgraph); ei != ei_end; ++ei) {
+		size_t componentNum = component[*ei];
+		if (componentNum + 1 > componentToVertexSet.size()) {
+			componentToVertexSet.resize(componentNum + 1);
+		}
+
+		vertex_t node1 = source(*ei, subgraph);
+		vertex_t node2 = target(*ei, subgraph);
+
+		if (art_points.find(node1) == art_points.end()) {
+			componentToVertexSet[componentNum].insert(subgraph[node1].indexOriginal);
+		}
+		if (art_points.find(node2) == art_points.end()) {
+			componentToVertexSet[componentNum].insert(subgraph[node2].indexOriginal);
+		}
+	}
+
+	size_t moleculeNum = 0;
+
+	// Remove components with size less than 1
+	for (auto&& vertexSet : componentToVertexSet) {
+		if (vertexSet.size() <= 1) {
+			continue;
+		}
+		for (auto&& vertex : vertexSet) {
+			vertexToComponent[vertex] = moleculeNum;
+		}
+		moleculeNum++;
+	}
+}
+
+int
+main(int argc, char* argv[])
+{
+
+	auto progname = "physlr-molecules";
+	int c;
+	int optindex = 0;
+	char* end = nullptr;
+	static int help = 0;
+	std::string separationStrategy = "bc";
+	bool verbose = false;
+	unsigned t = 1;
+	bool failed = false;
+	static const struct option longopts[] = {
+		{ "help", no_argument, &help, 1 },
+		{ "separation-strategy", required_argument, nullptr, 's' },
+		{ nullptr, 0, nullptr, 0 }
+	};
+	while ((c = getopt_long(argc, argv, "s:vt:", longopts, &optindex)) != -1) {
+		switch (c) {
+		case 0:
+			break;
+		case 's':
+			separationStrategy.assign(optarg);
+			break;
+		case 'v':
+			verbose = true;
+			break;
+		case 't':
+			t = strtoul(optarg, &end, 10);
+			break;
+		default:
+			exit(EXIT_FAILURE);
+		}
+	}
+	std::vector<std::string> infiles(&argv[optind], &argv[argc]);
+	if (argc < 1) {
+		printUsage(progname);
+		exit(EXIT_FAILURE);
+	}
+	if (help != 0) {
+		printVersion();
+		printUsage(progname);
+		exit(EXIT_SUCCESS);
+	} else if (infiles.empty()) {
+		printErrorMsg(progname, "missing file operand");
+		failed = true;
+	}
+	if (t > 1) {
+		std::cerr << "physlr-molecules does not support multithreading currently." << std::endl;
+		t = 1;
+	}
+	if (separationStrategy != "bc") {
```

Comment:
I'm not sure it makes sense to have the other separation strategies in the help page right now if they aren't actually supported in the code? We can update everything to be consistent as more are added in.

Reply:
Also doesn't look like this variable is ever used anywhere right now?

Reply:
Hmm I can remove the other mol strategy from the help, but Im going to keep the `separationStrategy` variable

Reply:
updated